pkg/didcomm/protocol/route/service.go

```diff
@@ -296,6 +326,91 @@ func (s *Service) handleForward(msg *service.DIDCommMsg) error {
 	return s.outbound.Forward(forward.Msg, dest)
 }
 
+// Register registers the agent with a router with the on connectionID passed in. The agent is
+// registered with the router and retrieves the router endpoint and routing keys. This function
+// throws an error if the agent is already registered against a router. This function internally
+// works as a sync function and works in-conjunction with handleGrant function. ie, The request
+// did-comm message is sent to the router and it waits for the grant response from the router
+// till timeout.
```

Comment:
We should be able to register against multiple routers.

Also, suggest some modifications to the text:

```
// Register this agent with the router on the other end of the connection identified by connectionID.
// This method blocks until a response is received from the router. 
```

Reply:
Created an issue https://github.com/hyperledger/aries-framework-go/issues/1076 and added a TODO to register against multiple routers. Also, updated the goDoc.